### ISSUE TITLE
new rule: concurrent index creation cannot be run in transaction

### ISSUE THREAD

**Issue by chdsbd** [sbdchd/squawk#332]
If you attempt to run a concurrent index creation in a transaction, Postgres will error. We should warn about this:

```sql
begin;
CREATE INDEX CONCURRENTLY IF NOT EXISTS idx_foo_bar ON account (foo);

-- Query 1 ERROR at Line 1: : ERROR:  CREATE INDEX CONCURRENTLY cannot run inside a transaction block
```

based on #331
